fortuna/prob_model/posterior/base.py

```diff
@@ -145,3 +153,58 @@ def save_state(
             checkpoint_path=checkpoint_path,
             keep=keep_top_n_checkpoints,
         )
+
+    def _sample_diag_gaussian(self, rng: Optional[PRNGKeyArray] = None, **kwargs) -> JointState:
```

Comment:
Maybe it's better to have this in a different class then the abstract Posterior. Only Laplace and ADVI need it for now right?

Reply:
Sounds good :)